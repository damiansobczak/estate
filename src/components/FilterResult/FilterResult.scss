@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.filterResult{
    display: none;
    flex-direction: column;
    opacity: 0;
    position: absolute;
    left: 0;
    right: 0;
    top: 60px;
    background-color: #fff;
    box-shadow: 0 20px 60px rgba(0, 0, 0, 0.2);
    border-radius: 4px;
    max-width: 1100px;
    margin: 0 auto;
    padding: 20px;
    overflow: hidden;

    @include breakpoint-down(lg) {
        margin: 0 30px;
    }

    &--visible {
        display: flex;
        animation-name: filterResultToggle;
        animation-duration: 0.2s;
        animation-direction: normal;
        animation-timing-function: ease-in-out;
        animation-fill-mode: both;
    }

    &__item {
        display: flex;
        align-items: center;
        width: 100%;
        margin-bottom: 15px;
        padding-bottom: 15px;
        border-bottom: 1px solid rgba($color-neutral-300, 0.3);
        position: relative;
        z-index: 0;
        transition: all 0.2s ease-in-out;

        &:before {
            content: "";
            height: 0;
            transition: all 0.2s ease-in-out;
        }

        &:hover {
            cursor: pointer;
            opacity: 0.7;

            &:before{
                content: "";
                position: absolute;
                left: -20px;
                margin: auto;
                width: 4px;
                height: 60px;
                background-color: $color-accent;
            }
        }

        &:last-child {
            border: 0;
            padding-bottom: 0;
            margin-bottom: 0;
        }
    }

    &__image {
        width: 60px;
        height: 60px;
        border-radius: 2px;
        object-fit: cover;
        margin-right: 20px;
    }

    &__col {
        margin-right: 20px;
        padding-right: 20px;
        min-width: 100px;
        max-width: 200px;
        overflow: hidden;

        &:first-of-type {
            min-width: 200px;
        }

        @include breakpoint-down(md) {
            &:nth-last-child(1){
                display: none;
            }
        }

        @include breakpoint-down(sm) {
            &:nth-last-child(2){
                display: none;
            }
        }

        @include breakpoint-down(xs) {
            &:nth-last-child(3){
                display: none;
            }
        }
    }

    &__title {
        width: 100%;
        font-weight: 600;
        color: $color-text-primary;
        margin: 4px 0;
        font-family: poppins, sans-serif;
        white-space: nowrap;
        max-width: 100%;
        text-overflow: ellipsis;
        overflow: hidden;
    }

    &__value {
        color: $color-text-secondary;
        white-space: nowrap;
        max-width: 100%;
        text-overflow: ellipsis;
        overflow: hidden;
    }
}

@keyframes filterResultToggle {
    from {
        top: 60px;
        opacity: 0;
    }
    to {
        top: 70px;
        opacity: 1;
    }
}