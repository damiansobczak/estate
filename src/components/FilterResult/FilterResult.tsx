import React from "react";
import "./FilterResult.scss";
import { FilterResultProps } from "./Interfaces/FilterResultProps";

class FilterResult extends React.Component<FilterResultProps> {
    render() {
        if (this.props.data) {
            return (
                <div className={`filterResult ${this.props.isShowing ? "filterResult--visible" : " "}`}>
                    {this.props.data.map((item: any) => (
                        <div className="filterResult__item" key={item.id}>
                            <img className="filterResult__image" src={item.better_featured_image != null ? item.better_featured_image.source_url : ''} alt={item.slug} />
                            <div className="filterResult__wrapper">
                                <div className="filterResult__title">{item.title.rendered}</div>
                                <div className="filterResult__desc">
                                    <div className="filterResult__price">{item.acf.cena}</div>
                                    <div className="filterResult__address">ul. {item.acf.ulica},</div>
                                    <div className="filterResult__space">{item.acf.metraz}</div>
                                </div>
                            </div>
                        </div>
                    ))}
                </div>
            );
        } else {
            return (
                <div className={`filterResult ${this.props.isShowing ? "filterResult--visible" : " "}`}>
                    Brak wyszuka≈Ñ
                </div>
            );
        }
    }
}

export default FilterResult;
