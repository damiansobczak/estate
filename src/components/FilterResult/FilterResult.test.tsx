import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import FilterResult from "./FilterResult";

const result = {
    isShowing: false,
    data: []
}

const result2 = {
    "isShowing": true,
    "data": [
        {
            "id": 42,
            "slug": "osiedle-babina",
            "link": "http://localhost:8888/estate/nieruchomosc/osiedle-babina/",
            "title": {
                "rendered": "Osiedle Babina"
            },
            "content": {
                "rendered": "\n<p>Przedstawiamy najnowszą ofertę biura nieruchomości. Inwestycja usytuowana w ustronnym zaciszu, z dala od miejskiego zgiełku. Poniżej znajdą Państwo wszelkie niezbędne informacje odnośnie oferty w tym: miejsce usytuowania, dodatkowe wyposażenie oraz całkowite koszty mieszkalne.</p>\n",
                "protected": false
            },
            "excerpt": {
                "rendered": "<p>Przedstawiamy najnowszą ofertę biura nieruchomości. Inwestycja usytuowana w ustronnym zaciszu, z dala od miejskiego zgiełku. Poniżej znajdą Państwo wszelkie niezbędne informacje odnośnie oferty w tym: miejsce usytuowania, dodatkowe wyposażenie oraz całkowite koszty mieszkalne.</p>\n",
                "protected": false
            },
            "categories": [
                3
            ],
            "better_featured_image": {
                "id": 83,
                "media_details": {
                    "width": 1400,
                    "height": 933,
                    "file": "2020/01/image-1.jpg",
                },
                "source_url": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-1.jpg"
            },
            "acf": {
                "ulica": "ul. Babina 20/3A",
                "metraz": "76m2",
                "liczba_pokoi": "3",
                "cena": "125,500zł",
                "typ_domu": "Kamienica",
                "liczba_pieter": "1",
                "miejsce_parkingowe": "1",
                "powierzchnia_dzialki": "100m2",
                "promocja": true,
                "wspolrzedne": {
                    "lat": "51.748378",
                    "lng": "19.464890"
                },
                "kaucja": "1500",
                "media": "300",
                "czynsz": "250",
                "atrybuty": {
                    "lazienka": "Prysznic, pralka",
                    "pokoje": "2, TV, stół, sofa",
                    "kuchnia": "Kuchenka, czajnik",
                    "balkon": "6m2, zastawa",
                    "parking": "Miejsce, szlaban",
                    "metraz": "26m2, taras"
                },
                "tlo": false
            }
        },
        {
            "id": 40,
            "slug": "osiedle-srebrne-laki",
            "link": "http://localhost:8888/estate/nieruchomosc/osiedle-srebrne-laki/",
            "title": {
                "rendered": "Osiedle Srebrne Łąki"
            },
            "content": {
                "rendered": "\n<p>Przedstawiamy najnowszą ofertę biura nieruchomości. Inwestycja usytuowana w ustronnym zaciszu, z dala od miejskiego zgiełku. Poniżej znajdą Państwo wszelkie niezbędne informacje odnośnie oferty w tym: miejsce usytuowania, dodatkowe wyposażenie oraz całkowite koszty mieszkalne.</p>\n",
                "protected": false
            },
            "excerpt": {
                "rendered": "<p>Przedstawiamy najnowszą ofertę biura nieruchomości. Inwestycja usytuowana w ustronnym zaciszu, z dala od miejskiego zgiełku. Poniżej znajdą Państwo wszelkie niezbędne informacje odnośnie oferty w tym: miejsce usytuowania, dodatkowe wyposażenie oraz całkowite koszty mieszkalne.</p>\n",
                "protected": false
            },
            "categories": [
                3
            ],
            "better_featured_image": {
                "id": 82,
                "source_url": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-2.jpg"
            },
            "acf": {
                "ulica": "ul. Srebrna",
                "metraz": "50m2",
                "liczba_pokoi": "4",
                "cena": "100,000zł",
                "typ_domu": "Dom jednorodzinny",
                "liczba_pieter": "2",
                "miejsce_parkingowe": "2",
                "powierzchnia_dzialki": "100m2",
                "promocja": true,
                "wspolrzedne": {
                    "lat": "51.748378",
                    "lng": "19.464890"
                },
                "kaucja": "2200",
                "media": "140",
                "czynsz": "230",
                "atrybuty": {
                    "lazienka": "Prysznic, pralka",
                    "pokoje": "2, TV, stół, sofa",
                    "kuchnia": "Kuchenka, czajnik",
                    "balkon": "6m2, zastawa",
                    "parking": "Miejsce, szlaban",
                    "metraz": "26m2, taras"
                },
                "tlo": false
            }
        }
    ]
}

describe("<FilterResult />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(<FilterResult {...result} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("shows info when no results", () => {
        const wrapper = shallow(<FilterResult {...result} />);
        expect(wrapper.find(".filterResult").prop("className")).not.toContain("filterResult--visible");
        expect(wrapper.find(".filterResult__title").html()).toContain("Brak ofert");
    });

    it("shows modal when available results", () => {
        const wrapper = shallow(<FilterResult {...result2} />);
        expect(wrapper.find(".filterResult").prop("className")).toContain("filterResult--visible");
    });

    it("shows multiple results", () => {
        const wrapper = shallow(<FilterResult {...result2} />);
        expect(wrapper.find(".filterResult__item").length).toEqual(2);
    });

    it("shows fields in item correctly", () => {
        const wrapper = shallow(<FilterResult {...result2} />);
        expect(wrapper.find(".filterResult__item").first().find(".filterResult__image").prop("src")).toContain("http://localhost:8888/estate/wp-content/uploads/2020/01/image-1.jpg");
        expect(wrapper.find(".filterResult__item").first().find(".filterResult__title").first().html()).toContain("Osiedle Babina");
        expect(wrapper.find(".filterResult__item").first().find(".filterResult__value").at(1).html()).toContain("125,500zł");
        expect(wrapper.find(".filterResult__item").first().find(".filterResult__value").at(2).html()).toContain("76m2");
        expect(wrapper.find(".filterResult__item").first().find(".filterResult__value").at(3).html()).toContain("3");
    });
});
