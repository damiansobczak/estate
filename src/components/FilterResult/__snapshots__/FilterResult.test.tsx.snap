// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterResult /> renders component correctly 1`] = `
<div
  className="filterResult  "
>
  <div
    className="filterResult__title"
  >
    Brak ofert
  </div>
</div>
`;
